airflow-core/src/airflow/models/deadline.py

```diff
@@ -90,3 +96,77 @@ def _determine_resource() -> tuple[str, str]:
     def add_deadline(cls, deadline: Deadline, session: Session = NEW_SESSION):
         """Add the provided deadline to the table."""
         session.add(deadline)
+
+
+class DeadlineTrigger(Enum):
```

Comment:
I am pretty sure we already had this conversation but I cannot remember where we settled. I find the term `DeadlineTrigger` confusing because `trigger` is already used in Airflow and can confuse users/developers. I would rather chose another term (I do not a proposal right now but i can comes later :)). WDYT?

Reply:
I know it has come up a bunch of times and didn't remember ever settling on a better name, but looking back int he dev list it looks like we settled on `DeadlineReference`.   I'll update the code.  Thanks for he reminder.